### PR TITLE
Redesigned manager login

### PR DESCRIPTION
**Pull request by Mark-H**
### What does it do?
This implements a new manager login screen. Designed by Rinze van der Brug, front-end by @christianseel, back-end/js by yours truly.

schermafbeelding 2018-02-09 om 23 49 24

The background and logo images can be changed with settings, and some new lexicons can be used to change the help text. 

### Why is it needed?

Modernize the manager login. 

### Related issue(s)/PR(s)

N/a

### CODE REVIEW

**Comment by christianseel** [comment]
The default background image was taken by myself in Engelberg, Switzerland during the MODX SnowUp. License is hereby granted for using it at the manager login page.

**Comment by Mark-H** [comment]
Just doing some quick accessibility testing using VoiceOver, and notice that we're hiding the forms with only a height declaration, causing the form to still be accessible to screenreaders. That situation is well explained at https://cbateman.com/blog/accessible-css-transitions/ and should be solvable by adding `visibility: none;` plus a (delayed) transition on visibility.

**Comment by digitalpenguin** [comment]
Looks great! I'm not so sure about the "feel free to login anytime" sentence though..

**Comment by gpsietzema** [comment]
Loving it :-) I know what you mean @digitalpenguin - any ideas as an alternative?

We should also approach this as a bigger thing though. I browsed a lot of (old) issues yesterday and a lot of them mentioned lexicons/language. This is a very nice start to making MODX a little friendlier though.

**Comment by digitalpenguin** [comment]
Not as yet unfortunately. I've been thinking about it though - it's a tough one! ;)

**Comment by sottwell** [comment]
Anything that makes the login page easier to customize will be an improvement. Modifying the login page shouldn't require a course in ExtJS and Smarty!

**Comment by christianseel** [comment]
Should we replace
> Remember me
with 
> Keep me logged in for 14 days
? (depending on the current lifetime settings)

**Comment by christianseel** [comment]
It's also fully responsive…
image 2018-02-10 at 3 33 11 pm

**Comment by rthrash** [comment]
Love, love, love this. Great work everyone—definitely more friendly!

Perhaps a single line of copy would work, including if it's your first time logging in along the lines of:

> Good (morning|afternoon|evening), log in to manage your site.

**Comment by rthrash** [comment]
One other question comes to mind, would it be straightforward to add 2FA to this? Most implementations take you to another screen with just a 2FA code box and a checkbox to remember this device for XX days.

(A\wesome photo @christianseel, too! 🥇)

**Comment by christianseel** [comment]
Now we also got the language switch back! Now with full native language strings instead of just lang codes.
bildschirmfoto 2018-02-10 um 18 04 33

File: core/lexicon/en/login.inc.php [hunk 1]
```diff
@@ -38,7 +38,7 @@
 $_lang['login_greeting_afternoon'] = '<strong>Good afternoon</strong>, welcome back!';
 $_lang['login_greeting_evening'] = '<strong>Good evening</strong>, welcome back!';
 $_lang['login_greeting_night'] = '<strong>Good night</strong>, welcome back!';
-$_lang['login_note'] = 'Feel free to login anytime.';
+$_lang['login_note'] = 'Please login to access the manager.';
```
Comment (rthrash) [inline core/lexicon/en/login.inc.php]: “Manager“ should be capitalized, and I believe it should be "log in" also.

Comment (christianseel) [reply core/lexicon/en/login.inc.php]: we did a vote on "login" vs "log in" and decided both work. I'm fine with both.

Comment (christianseel) [reply core/lexicon/en/login.inc.php]: `Manager` fixed. Ben, @Mark-H and @gpsietzema voted for `login` instead `log in`.

**Comment by wuuti** [comment]
I really like the idea to print how long the "Keep me logged in" cookie extension will be. So something like "Keep me logged in for the next 30 days on this device" would be nice. I now this is making the text longer, but that's way more transparent to the user (maybe we find a shorter text).

**Comment by christianseel** [comment]
Just for @wuuti 
bildschirmfoto 2018-02-10 um 21 16 18
(this recognizes the `session_cookie_lifetime` setting and outputs the lifetime accordingly)

File: core/lexicon/en/login.inc.php (hunk 1 above)
Comment (rthrash) [reply core/lexicon/en/login.inc.php]: Grammatically, "log in" is a verb. A "login" is a noun or adjective.

After you enter your login to log in, you've logged in, not loginned. You can ask a friend to log you in, but it'd sound weird if you tried to get them to login you.

:P

Comment (christianseel) [reply core/lexicon/en/login.inc.php]: Sounds ok for me. Updated!

**Comment by jaygilmore** [comment]
I absolutely love the overall treatment. 

It seems to me that a ghost button for the Forgot your Login is a) too close for to the login button, b) contains the word login and c) looks like a button. This can make it easier for an accidental click target because it's the last button in the view, and for people in a hurry, may just click whichever button. Convention generally has the Forgot Password/Login functionality triggered by a link. 

Here are a few examples: 
login mailchimp - email marketing made easy 2018-02-12 12-57-52
sign in to github github - private browsing 2018-02-12 12-55-24
dribbble - sign in - private browsing 2018-02-12 12-58-59

**Comment by jaygilmore** [comment]
Where to the help and language things go when scaled down for small viewports/mobile?
Also, what kind of element is the Language selector? It looks like a button. Should it be a select? Should it look like a select?

**Comment by christianseel** [comment]
Language: it’s a select and currently lacks a small arrow to indicate the drop down. Just didn’t have time. Help is a button.

**Comment by christianseel** [comment]
Screenshots after latest updates:
bildschirmfoto 2018-02-13 um 10 08 29
image 2018-02-13 at 10 21 48 am

**Comment by jonleverrier** [comment]
Great work - loving the new manager...

I can't help but feel the help and language select fields feel a little disjointed and out of context on desktop compared to mobile. The buttons on desktop look like they do something to the large background area.

**Comment by gpsietzema** [comment]
@jonleverrier Any suggestions?

**Comment by christianseel** [comment]
…we could move into the white area next to the logo? Through I find personally that this makes them less discoverable.

**Comment by Jako** [comment]
I prefer the same location as on the mobile screen (top of the modx logo).

**Comment by jonleverrier** [comment]
@gpsietzema I think the mobile version has the right hierarchy of information (what @Jako said). 

Great work guys!

**Comment by rthrash** [comment]
I'm a fan of the mobile layout as well for the buttons/language selector on the desktop also.

**Comment by Mark-H** [comment]
I'd say this is ready to get merged in. Per earlier discussions (primarily in Slack/in person at the SnowUp) this should be going into 3.x instead of 2.x, so it's part of the bigger MODX3 effort happening right now. 

Regarding things like the button positions, we can continue to make changes in separate PRs. It's a pretty massive improvement, so it's good that there's discussion among design-minded folks about the fine details, so let's continue that in an issue or other medium and iterate on what's there.

**Comment by SnowCreative** [comment]
OK, a small beef here. "Login" is not a verb. It's a noun (and sometimes used as an adjective). "Log in" is correct for use as a verb, as in Christian's latest screen shot above ("Log in to access the manager"). So the button should say "Log In", like it does in the Mailchimp screen shot. If "Login" were a verb, then you would say "Keep me loginned for 2 weeks"!

**Comment by ghost** [comment]
@rainbowtiger - give it up. Conference and reference also aren't verbs. One confers with someone or has a conference, and one refers to something or makes a reference, but I haven't seen either of them used properly for decades.

**Comment by Mark-H** [comment]
PRs to further improve the login are welcome. :)

**Comment by SnowCreative** [comment]
"Reference" has officially been a verb for a while now, and even "conference" is listed as a verb in some dictionaries, although that's more of a stretch. "Login" is not listed as a verb anywhere. Furthermore, when people use "login" as a verb, they pronounce it "log in", with equal stress on both words and a slight separation between them, rather than just accenting the first syllable as they do with "login" as a noun. The misuse of "it's" as a possessive pronoun is also rampant right now, but that doesn't mean we should just accept that and allow its usage. And, I doubt that any dictionaries or grammar texts will ever change to include "it's" as a pronoun rather than a contraction. 

My opinion is that if we want to make something look as professional as possible, using correct grammar helps toward that end.

P.S. Sorry, Mark, I don't know how to do PRs. Maybe one of these days I'll get around to learning how that works.

**Comment by Mark-H** [comment]
This PR was already merged so is not the place to discuss it. I doubt anyone would be opposed to improving the English grammar, so please do send a PR (or create a new issue if you're not comfortable with git) for the correction.

**Pull request merged**